src/ApproximateTNContraction/contract/mincut_tree.jl

```diff
@@ -0,0 +1,266 @@
+    return inds_binary_tree!(tng, inds; kwargs...)
+  end
+  inds_groups = [get_sub_tree(inds) for inds in inds_groups]
+  if length(inds_groups) <= 2
+    return inds_groups
+  end
+  return inds_binary_tree!(tng, inds_groups; kwargs...)
+end
+
+function inds_binary_tree!(tng::TensorNetworkGraph, outinds::Vector; algorithm="mincut")
+  @assert algorithm in ["mincut", "mincut-mps", "mps"]
+  @assert all(ind -> ind in keys(tng.out_edge_dict), outinds)
+  if algorithm == "mincut"
+    return mincut_inds!(tng, outinds)
+  elseif algorithm == "mincut-mps"
+    inds_tree = mincut_inds!(tng, outinds)
+    linear_tree = linearize(inds_tree, tng)
+    out_inds = linear_tree[1]
+    for i in 2:length(linear_tree)
+      out_inds = [out_inds, linear_tree[i]]
+    end
+    return out_inds
+  elseif algorithm == "mps"
+    return line_to_tree(inds_linear_order!(tng, outinds))
+  end
+end
+
+function inds_linear_order(
+  network::Vector{ITensor}, outinds::Union{Nothing,Vector{<:Index}}
+)
+  if outinds == nothing
+    outinds = noncommoninds(network...)
+  end
+  if length(outinds) == 1
+    return outinds
+  end
+  tng = TensorNetworkGraph(network, outinds)
+  grouped_uncontracted_inds = [[i] for i in outinds]
+  return inds_linear_order!(tng, grouped_uncontracted_inds)
+end
+
+function inds_linear_order!(tng::TensorNetworkGraph, outinds::Vector)
+  @assert length(outinds) >= 1
+  # base case here, for the case length(outinds) == 2, we still need to do the update
+  if length(outinds) == 1
+    return outinds[1]
+  end
+  if length(outinds) == 2
+    return outinds
+  end
+  new_edge, minval = new_edge_mincut(tng, collect(powerset(outinds, 1, 1)))
+  new_edge = new_edge[1]
+  # outinds = update!(tng, outinds, new_edge, minval)
+  # first_ind = new_edge
+  linear_order = [new_edge]
+  while length(outinds) > 2
+    splitinds = [[new_edge, i] for i in outinds if i != new_edge]
+    new_edge, minval = new_edge_mincut(tng, splitinds)
+    outinds = update!(tng, outinds, new_edge, minval)
+    # first_ind = new_edge
+    push!(linear_order, new_edge[2])
+  end
+  splitinds = [[new_edge, i] for i in outinds if i != new_edge]
+  @assert length(splitinds) == 1
+  push!(linear_order, splitinds[1][2])
+  return linear_order
+end
+
+function inds_binary_tree(
+  network::Vector{ITensor}, outinds::Union{Nothing,Vector{<:Index}}; algorithm="mincut"
+)
+  if outinds == nothing
+    outinds = noncommoninds(network...)
+  end
+  if length(outinds) == 1
+    return outinds
+  end
+  if algorithm == "sequential-mps"
+    out_inds = [outinds[1]]
+    for i in 2:length(outinds)
+      out_inds = [out_inds, [outinds[i]]]
+    end
+    return out_inds
+  end
+  tng = TensorNetworkGraph(network, outinds)
+  grouped_uncontracted_inds = [[i] for i in outinds]
+  return inds_binary_tree!(tng, grouped_uncontracted_inds; algorithm=algorithm)
+end
+
+#TODO: rewrite this function
+#TODO: pick one end deterministically
+function linearize(inds_tree::Vector, tng::TensorNetworkGraph)
```

Comment:
@LinjianMa could you explain what this function is doing? I'm curious if it can be written more simply in terms of recent functionality I added in https://github.com/mtfishman/NamedGraphs.jl/pull/20, which should all work for `AbstractITensorNetwork` subtypes now. If not, it would be helpful to identify if there are general graph functionality we could make use of or factorize out from this function.

Reply:
this is a retired and unused function, I have already removed it